llama.cpp

```diff
@@ -1805,10 +1865,13 @@ struct llama_tokenizer {
         size_t offs = 0;
         while (offs < text.size()) {
             llama_sp_symbol sym;
-            size_t char_len = std::min(text.size() - offs, utf8_len(text[offs]));
+            size_t len = utf8_len(text[offs]);
+            if (len == 0) {
+                len = utf8_len(text[offs]);
+            }
```

Comment:
This doesn't look right

Reply:
Correct, this is an artefact from using a trie in an earlier commit, will remove.

Reply:
Done.